### PR TITLE
Use hot-shots native childClient

### PR DESCRIPTION
**Pull request by hknustwmf**
Replaced statsd client instances creation with childClient method calls in makeChild. Upped the version number to 2.6.10.

Bug:  [T203617](https://phabricator.wikimedia.org/T203617)

### CODE REVIEW

File: lib/docker.js [hunk 1]
```diff
@@ -212,7 +214,7 @@ function createDockerFile() {
     }
 
     if (!opts.deploy) {
-        contents += `RUN mkdir /opt/service\nADD . /opt/service\nWORKDIR /opt/service\nRUN ${npmCommand} install && npm dedupe\n`;
+        contents += `RUN mkdir /opt/service\nADD . /opt/service\nWORKDIR /opt/service\nRUN rm -rf node_modules && ${npmCommand} install && npm dedupe\n`;
```
Comment (d00rman) [inline lib/docker.js]: We don't do `rm -rf node_modules` here on purpose: we want developers to be able to quickly spin up their service

File: lib/docker.js [hunk 2]
```diff
@@ -595,7 +600,7 @@ function main(options, configuration) {
         // set the defaults
         targets = {
             debian: 'debian:jessie',
-            ubuntu: 'ubuntu:14.04'
+            ubuntu: 'ubuntu:18.04'
```
Comment (d00rman) [inline lib/docker.js]: Let's not do changes that don't belong conceptually in this PR. Also, the last Ubuntu version we supported in prod was 14.04 so better to keep it there (even though we don't use Ubuntu in production any more at all).

Comment (hknustwmf) [reply lib/docker.js]: Reverted back

File: lib/docker.js (hunk 1 above)
Comment (hknustwmf) [reply lib/docker.js]: Removed

**Comment by Pchelolo** [comment]
first things first - let's not mix completely unrelated things into a single pull request. The docker.js changes are good, they just entirely do not belong here.

File: lib/statsd.js [hunk 3]
```diff
@@ -95,8 +105,6 @@ function makeStatsD(options, logger) {
         if (typeof originalClose === 'function') {
             originalClose.apply(statsd);
         }
-        statsd._children.forEach(child => child.close());
-        statsd._children = [];
     })(statsd.close);
```
Comment (Pchelolo) [inline lib/statsd.js]: We don't need this anymore at all. The whole redefinition of the 'close' method.

File: lib/statsd.js [hunk 4]
```diff
@@ -71,15 +72,24 @@ function makeStatsD(options, logger) {
...
+        const child = statsd.childClient(childOptions);
+
+        // Attach normalizeName to make the childClient instance backwards-compatible
+        // with the previously used StatsD instance
+        child.normalizeName = normalizeName;
+
+        // Also add a small utility to send a delta given a startTime
+        child.endTiming = function endTiming(names, startTime, samplingInterval) {
+            return this.timing(names, Date.now() - startTime, samplingInterval);
+        };
```
Comment (Pchelolo) [inline lib/statsd.js]: This code is duplicated on line 99. We can just make it a const endTiming and then bind it to an appropriate 'this'.

Alternatively event better, we can make a function that will apply the mixin to the original hot-shots statsd and share it for both the root and the child.

**Comment by Pchelolo** [comment]
package-lock.js is here.

**Comment by Pchelolo** [comment]
Did you test it with RB?

**Comment by Pchelolo** [comment]
While you're on this repo, couble you make a separate PR to remove mocha-eslint and eslint-config-node-services? I forgot to add this repo into the list

File: lib/statsd.js (hunk 3 above)
Comment (hknustwmf) [reply lib/statsd.js]: Removed.

File: lib/statsd.js (hunk 4 above)
Comment (hknustwmf) [reply lib/statsd.js]: Added mixin

**Comment by hknustwmf** [comment]
> Did you test it with RB?

Yes. I ran restbase to test the service-runner changes

**Comment by Pchelolo** [comment]
The description of the PR is not correct - this one does not touch the Docker.js

**Comment by d00rman** [comment]
Published as [v2.6.10](https://github.com/wikimedia/service-runner/tree/v2.6.10)

**Pull request merged**